apps/accounts/views.py

```diff
@@ -89,8 +103,17 @@ def post(self, request, user, identity):
             return Response({'error': 'That account already exists.'},
                             status=422)
         else:
-            user = UserProfile.objects.create_user(
-                email=identity['email'], username=None, fxa_id=identity['uid'])
-            log.info('Created user {} from FxA'.format(user))
-            login(request, user)
+            user = register_user(request, identity)
             return Response({'email': user.email})
+
+
+class AuthorizeView(APIView):
+
+    @waffle_switch('fxa-auth')
+    @with_user
+    def get(self, request, user, identity):
+        if user is None:
+            register_user(request, identity)
+        else:
+            login_user(request, user, identity)
+        return HttpResponseRedirect('/')
```

Comment:
I was thinking that too but it would need to go on the state and we're currently just passing `foo` for the state. There are two other bugs to start putting stuff in the state. I think I'll file another bug for now to do after the user id in state bug.